Task 24

monicayun opened the pull request:
added test with 3 ideals
@sonjapetrovic @parkerjoncus @dwilburne

monicayun committed:
copy pasted function

monicayun committed:
added colons

monicayun committed:
added ring

Commit:
deletion of file and export

deleted unnecessary file code and added name into export

monicayun committed:
changed indentation

monicayun committed:
added documentation block

Commit:
Added tests for pdim

added 4 tests for pdim. 1 with zero ideal, one with basis, and two set.

Commit:
added 1 more test

added test with 3 ideals

sonjapetrovic committed:
added useful text after "end"

for copy+paste purposes! :)

sonjapetrovic committed:
edits to method + doc (needs review)

Sondzus commented:
I made 2 edits: 
- added useful text after "end" so I can copy+paste when installing
- edited both method and documentation to streamline. But we are missing some functionality:
@monicayun @parkerjoncus  include stdev?

dkosmas commented:
I got an error trying to install this package:
error: mutable unexported unset symbol(s) in package RandomMonomialsIdeals: 'ideals'
currentString:1:5-1:11: here is the first use of 'ideals'

Sondzus commented:
@dkosmas that might have been my fault, I'll take a second look tomorrow.

Sondzus commented:
Also more work is required here, the branch isn't complete yet. Thanks for the heads up on the error.

Commit:
added stdDev

Added stdDev and other parts to function like the option to to see tally

parkerjoncus commented:
I just added stdDev and the option, but I am also now getting the error for first use of ideals and I am not sure where it is coming from

Sondzus commented:
let me take a look at ideals error.

sonjapetrovic committed:
I found the "ideals" error.

fixed it.

But there are 2 more errors now - not sure where they come from

Sondzus commented:
@monicayun @parkerjoncus  it was a silly "s" i typed in the documentation that caused the "ideals" error before. I fixed it. but now i have another issue - pdimStats doesn't run properly (there's an error in documentation). I commented out the 3 lines that cause the error; try it out. 

Here is what I get:


i1 : L=randomMonomialSet(3,3,1.0);

i2 : R=ring(L#0);

i3 : ideals = {monomialIdeal(R_0^3,R_1,R_2^2), monomialIdeal(R_0^3, R_1, R_0*R_2)};

i4 : listOfIdeals = idealsFromGeneratingSets(randomMonomialSets(4,3,1.0,3));
There are 3 ideals in this sample.
Of those, 0 were the zero ideal.

i5 : pdimStats(listOfIdeals)
stdio:5:1:(3): error: no method for binary operator * applied to objects:
--            1
--            - (of class QQ)
--            3
--      *     (12, RR) (of class Sequence)

Commit:
fixed running error, added stdDev to tests

fixed the error from calling function and added check for the stdDev on
the test examples

parkerjoncus commented:
Alright, the run error is fixed, and I added the standard deviation checks into the tests.

sonjapetrovic committed:
changed a doc example

- easier to decipher the example (seeing ideals, instead of guessing
what they are, is easier).
- added [pdimStats,ShowTally] to doc.
- added a QUESTION to ShowTally doc node - as a comment - I don’t know
why it’s formatted the way it is?

Sondzus commented:
The package installs and tests run successfully. 
I made some changes to documentation to try to streamline it / make it more readable by the user.

sonjapetrovic committed:
let the user see the ideals in doc

This way it is easy to tell (for an algebraist) whether the answer
(4,0) makes any sense at all. It does.

Sondzus commented:
OK I removed the " ; " from the documentation - let the user see the ideals.
Also are these examples in doc as good as they can be? I mean, the ideals we generate are pretty boring: both times it's just 3 copies of 1 ideal, (x1,x2,x3,x4). Maybe at least the 2nd example can be something different. Type up some monomials by hand to get some different pdim values in the output...
@monicayun @parkerjoncus 

Agreed, @dwilburne @sdgpfo ?

Commit:
added another example

added anther example to documentation that is more interesting and it
shows the tally. Added SeeAlso.

parkerjoncus commented:
Not sure if we need the two examples of the x1,x2,x3,x4 where they are generated using other built in functions

Commit:
Changed so a,b,c instead of x1,x2,x3

Edited example so more user friendly

Commit:
small edit

change of wording

Commit:
Deletion of example

Deleted a redundant example and edit to spelling

Commit:
Merge branch 'master' into Task-24

# Conflicts:
#	RandomMonomialIdeals.m2

Commit:
merge fixes

Commit:
Revert "merge fixes"

This reverts commit af64c314263c8e4ae9392a2fb10091260eed7e41.

parkerjoncus committed:
Merge branch 'master' into Task-24
